be/src/olap/storage_engine.cpp

```diff
@@ -685,6 +693,8 @@ OLAPStatus StorageEngine::_start_trash_sweep(double* usage) {
     // clean unused rowset metas in OlapMeta
     _clean_unused_rowset_metas();
 
+    _trash_sweep_lock.unlock();
```

Comment:
The method may be returned before you unlock this lock.
you can use `src//util/mutex.h` to unlock automatically when deconstructing.

Reply:
> 
> 
> The method may be returned before you unlock this lock.
> you can use `src//util/mutex.h` to unlock automatically when deconstructing.

I use unique_lock to fix it.